src/pywinctl/_pywinctl_linux.py

```diff
@@ -609,19 +609,24 @@ def hide(self, wait: bool = False) -> bool:
             time.sleep(WAIT_DELAY * retries)
         return not self._isMapped
 
-    def activate(self, wait: bool = False):
+    def activate(self, wait: bool = False, user: bool = True) -> bool:
         """
         Activate this window and make it the foreground (focused) window
 
         :param wait: set to ''True'' to wait until action is confirmed (in a reasonable time lap)
+        :param user: ''True'' indicates a direct user request, as required by some WMs to comply.
         :return: ''True'' if window activated
         """
         if "arm" in platform.platform():
             EWMH.setWmState(self._hWnd, ACTION_UNSET, STATE_BELOW, STATE_NULL)
             EWMH.display.flush()
             EWMH.setWmState(self._hWnd, ACTION_SET, STATE_ABOVE, STATE_FOCUSED)
         else:
-            EWMH.setActiveWindow(self._hWnd)
+            # https://specifications.freedesktop.org/wm-spec/wm-spec-latest.html#sourceindication
+            source = 2 if user else 1
+            EWMH._setProperty('_NET_ACTIVE_WINDOW',  # noqa : W0212
```

Comment:
See? I told you that accessing a private method would give us some headaches :-P

Not sure why pyright and mypy didn't honor the `# noqa : W0212` remark.
Or maybe we just need to add `EWMH._setProperty` to your ewmh typing file?

Reply:
HAHAHAHA! Yes, you told me... but there was no other option!

I'm having my own hell trying to commit other changes. I have done more than 30 commits this evening, without success... still trying. As I told you, all this about typing is like black magic to me. I'm not saying it is not good, or necessary, or useful or even highly recommendable... but trust me that right now even the last of my cells deeply hate it.

Reply:
Well, using typing is one thing, _requiring_ full 100% coverage for _all_ methods, including external libraries, _that's_ insane! lol